lib/new_relic/agent/instrumentation/active_job.rb

```diff
@@ -26,6 +26,19 @@
       NewRelic::Agent::PrependSupportability.record_metrics_for(ActiveJob::Base)
     end
   end
+
+  executes do
+    if defined?(ActiveSupport) &&
+        ActiveJob.respond_to?(:gem_version) &&
+        ActiveJob.gem_version >= Gem::Version.new('6.0.0') &&
+        !NewRelic::Agent.config[:disable_activejob] &&
+        !NewRelic::Agent::Instrumentation::ActiveJobSubscriber.subscribed?
+      NewRelic::Agent.logger.info('Installing notifications based ActiveJob instrumentation')
```

Comment:
I apologize if this pattern has been around for a few PRs and I'm just noticing it. Why did you choose to put an `if` statement inside the `executes` block, instead of using a `depends_on` block? Since we have two different types of instrumentation for Active Job, should the registration of the instrumentation be in different files?

Reply:
Thanks for taking a look. You didn't miss this pattern anywhere else, and it's definitely representative of a compromise that bears further discussion. What we have here is:

- Existing ActiveJob instrumentation with existing `depends_on` checks
- New functionality that requires new `depends_on` type checks
- The new functionality still requires all of the existing `depends_on` checks
- The new functionality should use the existing config param
- The existing functionality is usable for older Rails apps, but the new stuff requires v6+

I had a choice to create a second instrumentation file. The drawback there is that I would have to either have the new file reference the existing one or create a third shared file or module that both instrumentation halves reference to be able to share the same checks and config setting.

Given that I wanted all of this functionality to collectively represent ActiveJob instrumentation that is all on or all off, and how small the new instrumentation's checks are, I made the call to put a conditional inside of an `executes` block.

I'm open to feedback on using a separate file, using a separate config option, and anything else.

Reply:
Thanks for breaking down the compromise. I think this works here and can be the topic of discussion in some other forum. We've talked before about standardizing the number of depends_on/executes blocks and I think this is a good case to consider when we have that discussion.